src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -982,16 +985,59 @@ def customizePerJob(self, job):
 
         # Performance estimates
         if job.get('estimatedJobTime'):
-            jdl.append('+MaxWallTimeMins = %d\n' % int(job['estimatedJobTime'] / 60.0))
+            estimatedMins = int(job['estimatedJobTime'] / 60.0)
+            origCores = int(job.get('numberOfCores', 1))
+            estimatedMinsSingleCore = estimatedMins*origCores
+            # For now, assume a 15 minute job startup overhead -- condor will round this up further
+            jdl.append('+EstimatedSingleCoreMins = %d\n' % estimatedMinsSingleCore)
+            jdl.append('+OriginalMaxWallTimeMins = %d\n' % estimatedMins)
+            jdl.append('+MaxWallTimeMins = WMCore_ResizeJob ? (EstimatedSingleCoreMins/RequestCpus + 15) : OriginalMaxWallTimeMins\n')
+            
         if job.get('estimatedMemoryUsage'):
-            jdl.append('request_memory = %d\n' % int(job['estimatedMemoryUsage']))
+            baseMem = int(job['estimatedMemoryUsage'])
+            jdl.append('+OriginalMemory = %d\n' % baseMem)
+            jdl.append('+RequestMemory = OriginalMemory + %d * (WMCore_ResizeJob ? (RequestCpus-OriginalCpus) : 0)\n' % self.extraMem)
+
         if job.get('estimatedDiskUsage'):
             jdl.append('request_disk = %d\n' % int(job['estimatedDiskUsage']))
 
-        # Set up JDL for multithreaded jobs
-        if job.get('numberOfCores', 1) > 1:
-            jdl.append('machine_count = 1\n')
-            jdl.append('request_cpus = %s\n' % job.get('numberOfCores', 1))
+        # Set up JDL for multithreaded jobs.
+        # By default, RequestCpus will evaluate to whatever CPU request was in the workflow.
+        # If the job is labelled as resizable, then the logic is more complex:
+        #  - If the job is running in a slot with N cores, this should evaluate to N
+        #  - If the job is being matched against a machine, match all available CPUs, provided
+        #    they are between min and max CPUs.
+        #  - Otherwise, just use the original CPU count.
+
+        jdl.append('machine_count = 1\n')
+        minCores = int(job.get('minCores', 1))
```

Comment:
@bbockelm as discussed in todays meeting, we shouldn't have a full range of resizing from multicore to 1 core.
So the initial idea is to resize down to half of the original number of cores (8 cores job should run with 4 cores, at least).
Can you please change this line to something like
`minCores = int(job.get('minCores', max(origCores/2, 1)))`